udata_recommendations/commands.py

```diff
@@ -18,29 +21,48 @@ def recommendations():
     pass
 
 
-def clean_datasets_recommendations():
-    log.info('Cleaning up dataset recommendations...')
+def clean_datasets_recommendations(source):
     datasets = Dataset.objects.filter(**{
-        'extras__recommendations__exists': True,
-    }).update(**{
-        'unset__extras__recommendations': True,
+        'extras__recommendations:sources__contains': source,
     })
-    success('Cleaned up %s dataset(s).' % datasets)
 
+    for dataset in datasets:
+        new_sources = set(
+            [s for s in dataset.extras['recommendations:sources'] if s != source]
+        )
+
+        # Dataset had only this source: delete all recommendations
+        if len(new_sources) == 0:
+            dataset.update(**{
+                'unset__extras__recommendations': True,
+                'unset__extras__recommendations:sources': True,
+            })
+            continue
+
+        # Dataset had at least one other source, keep only other sources
+        new_recommendations = [
+            r for r in dataset.extras['recommendations']
+            if r['source'] != source
+        ]
+        dataset.extras['recommendations:sources'] = list(new_sources)
+        dataset.extras['recommendations'] = new_recommendations
+        dataset.save()
+
+    log.info(f"Cleaned up {len(datasets)} dataset(s)")
+
+
+def get_recommendations_data(url):
+    response = requests.get(url, timeout=10)
+    response.raise_for_status()
+    data = response.json()
+
+    with importlib.resources.path('udata_recommendations', 'schema.json') as p:
+        schema_path = p
```

Comment:
Did not find a nicer way, I'm open to anything else

Reply:
`importlib` is fine 👌 

Why not nest both `with` and get rid of the `schema_path = p` line?